packages/engine/lib/stateful/src/agent/arrow/array.rs

```diff
@@ -76,11 +74,11 @@ impl IntoRecordBatch for &[&Agent] {
     }
 
     fn to_agent_batch(&self, schema: &AgentSchema) -> Result<RecordBatch> {
-        let mut cols = Vec::with_capacity(schema.arrow.fields().len());
+        let mut cols = Vec::with_capacity(schema.arrow.fields.len());
 
-        for field in schema.arrow.fields() {
-            // If `name` isn't cloned, Rust wants schema to have longer lifetime.
-            let name = field.name().clone();
+        for field in schema.arrow.fields.clone() {
```

Comment:
What is the type of `fields`, `Vec`? Is cloning needed here or can it be done by reference?